FinSearchUnified/Tests/Bundle/SearchBundleFindologic/FacetHandler/TextFacetHandlerTest.php

```diff
@@ -0,0 +1,360 @@
+    }
+
+    public function testFacetModeForBooleanResult()
+    {
+        $data = '<?xml version="1.0" encoding="UTF-8"?><searchResult></searchResult>';
+        $filter = new SimpleXMLElement($data);
+
+        $facet = new ProductAttributeFacet(
+            'field',
+            ProductAttributeFacet::MODE_BOOLEAN_RESULT,
+            'field',
+            'Field'
+        );
+
+        $facetHandler = new TextFacetHandler();
+        $result = $facetHandler->generatePartialFacet($facet, new Criteria(), $filter);
+
+        $this->assertNull($result);
+    }
+
+    /**
+     * @dataProvider valueListDataProvider
+     *
+     * @param array $filterData
+     * @param ValueListFacetResult $facetResult
+     * @param ConditionInterface|null $condition
+     */
+    public function testFacetModeForListFacetResult(
+        array $filterData,
+        ValueListFacetResult $facetResult,
+        ConditionInterface $condition = null
+    ) {
+        $facet = new ProductAttributeFacet(
+            'vendor',
+            ProductAttributeFacet::MODE_VALUE_LIST_RESULT,
+            'vendor',
+            'Manufacturer'
+        );
+        $criteria = new Criteria();
+        if ($condition !== null) {
+            $criteria->addCondition($condition);
+        }
+
+        $data = '<?xml version="1.0" encoding="UTF-8"?><searchResult></searchResult>';
+        $filter = new SimpleXMLElement($data);
+        foreach ($filterData as $key => $value) {
+            if (is_array($value)) {
+                $items = $filter->addChild('items');
+                foreach ($value as $itemData) {
+                    $item = $items->addChild('item');
+                    foreach ($itemData as $k => $v) {
+                        $item->addChild($k, $v);
+                    }
+                }
+            } else {
+                $filter->addChild($key, $value);
+            }
+        }
+
+        $facetHandler = new TextFacetHandler();
+        $result = $facetHandler->generatePartialFacet($facet, $criteria, $filter);
+
+        $this->assertEquals($facetResult, $result);
+    }
+
+    /**
+     * @dataProvider radioListDataProvider
+     *
+     * @param array $filterData
+     * @param RadioFacetResult $facetResult
+     * @param ConditionInterface|null $condition
+     */
+    public function testFacetModeForRadioFacetResult(
+        array $filterData,
+        RadioFacetResult $facetResult,
+        ConditionInterface $condition = null
+    ) {
+        $facet = new ProductAttributeFacet(
+            'vendor',
+            ProductAttributeFacet::MODE_RADIO_LIST_RESULT,
+            'vendor',
+            'Manufacturer'
+        );
+        $criteria = new Criteria();
+        if ($condition !== null) {
+            $criteria->addCondition($condition);
+        }
+
+        $data = '<?xml version="1.0" encoding="UTF-8"?><searchResult></searchResult>';
+        $filter = new SimpleXMLElement($data);
+        foreach ($filterData as $key => $value) {
```

Comment:
Tests should be simple, this is way to complex. We need a test to test the test if it is to complex.